src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -130,32 +130,27 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
         retrieveSchemaFromApoc(query, params)
       } catch {
         case e: ClientException =>
-          e.code match {
-            case "Neo.ClientError.Procedure.ProcedureNotFound" => {
-              // TODO get back to Cypher DSL when rand function will be available
-              val query = s"""MATCH (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS}:${options.relationshipMetadata.source.labels.map(_.quote()).mkString(":")})
-                |MATCH (${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS}:${options.relationshipMetadata.target.labels.map(_.quote()).mkString(":")})
-                |MATCH (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS})-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${options.relationshipMetadata.relationshipType}]->(${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS})
-                |RETURN ${Neo4jUtil.RELATIONSHIP_ALIAS}
-                |ORDER BY rand()
-                |LIMIT ${options.schemaMetadata.flattenLimit}
-                |""".stripMargin
-              val params = Collections.emptyMap[String, AnyRef]()
-              retrieveSchema(query, params, { record => record.get(Neo4jUtil.RELATIONSHIP_ALIAS).asRelationship.asMap.asScala.toMap })
-            }
-          }
+          // TODO get back to Cypher DSL when rand function will be available
```

Comment:
Please add a log warn message that explains that we switch to the query resolution.
Do the same for nodes.

Reply:
same as above